### PR TITLE
Support optional message in OutOfGasException

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/exceptions.h [hunk 1]
```diff
@@ -28,7 +28,12 @@ namespace HeraVM {
 
 class OutOfGasException : public std::exception {
 public:
-  const char* what() const noexcept override { return "Out of gas."; }
+  explicit OutOfGasException(std::string const& _msg = "Out of gas."):
+    msg(std::move(_msg))
```
Comment (chfast) [inline src/exceptions.h]: Nope. You cannot move a `const` object (this still will call the copy constructor not the move constructor).
https://www.youtube.com/watch?v=ZKaoR3dP9uM

The canonical way would be:
```
explicit OutOfGasException(std::string _msg = "Out of gas."):
    msg(std::move(_msg))
```

File: src/exceptions.h [hunk 2]
```diff
@@ -28,7 +28,12 @@ namespace HeraVM {
...
+  {}
+  const char* what() const noexcept override { return msg.c_str(); }
+private:
+  std::string msg;
```
Comment (chfast) [inline src/exceptions.h]: If you don't plan to have runtime created messages, you can just keep the reference to a string literal.

File: src/exceptions.h (hunk 1 above)
Comment (axic) [reply src/exceptions.h]: It's a typo, there were multiple iterations and added the move at the last minute.

File: src/exceptions.h (hunk 2 above)
Comment (axic) [reply src/exceptions.h]: See the next PR depending on this for the reason.

File: src/exceptions.h (hunk 1 above)
Comment (axic) [reply src/exceptions.h]: Fixed.

**Pull request merged**